src/init/network.rs

```diff
@@ -0,0 +1,281 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::str;
+use std::{cmp, fs, io};
+
+use ipnetwork::IpNetwork;
+use ipnetwork::Ipv6Network;
+
+use rtnetlink::Handle;
+use netlink_packet_route::rtnl::link::nlas::Nla;
+
+#[derive(Eq, PartialEq)]
+enum IpType {
+    V6,
+    V4,
+}
+
+fn get_ip_type(ip: &Vec<u8>) -> Option<IpType> {
+    if ip.len() == 4 {
+        return Some(IpType::V4);
+    } else if ip.len() == 16 {
+        return Some(IpType::V6);
+    }
+    None
+}
+
+pub fn get_sriov_capabilities(iface: &str) -> Result<String, io::Error> {
+    fs::read_to_string(format!(
+        "/sys/class/net/{}/device/sriov_totalvfs",
+        iface
+    ))
+}
+
+async fn set_link_up(handle: Handle, iface: &str) -> Result<(), anyhow::Error> {
+    let mut links = handle.link().get().match_name(iface.to_string()).execute();
+
+    if let Some(link) = links.try_next().await? {
+        handle.link().set(link.header.index).up().execute().await?
+    } else {
+        warn!("iface '{}' not found", iface);
+    }
+    Ok(())
+}
+
+pub async fn add_address(
+    iface: &str,
+    ip: IpNetwork,
+    handle: rtnetlink::Handle,
+) -> Result<(), anyhow::Error> {
+    let mut links = handle.link().get().match_name(iface.to_string()).execute();
+
+    if let Some(link) = links.try_next().await? {
+        handle
+            .address()
+            .add(link.header.index, ip.ip(), ip.prefix())
+            .execute()
+            .await?
+    }
+
+    Ok(())
+}
+
+// Create max(limit, max possible sriov for given iface) sriov devices for the given iface
+pub fn setup_sriov(iface: &str, limit: u16) {
+    if limit == 0 {
+        return;
+    }
+
+    let sriov_totalvfs = match get_sriov_capabilities(iface){
+        Ok(val) => val,
+        Err(e) => {
+            error!("{}", e);
+            return
+        }  
+    };
+
+    let sriov_totalvfs = match sriov_totalvfs.trim_end().parse::<u16>(){
+        Ok(val) => val,
+        Err(e) => {
+            error!("{}", e);
+            return
+        }  
+    };
+
+    let num = cmp::min(limit, sriov_totalvfs);
+
+    fs::write(
+        format!("/sys/class/net/{}/device/sriov_numvfs", iface),
+        num.to_string(),
+    )
+    .expect("Unable to write file");
+}
+
+pub async fn init_iface(
+    handle: rtnetlink::Handle,
+    ipv6: Ipv6Network,
+    iface: &str,
+    num_sriov: u16,
+) {
+    if let Err(e) =
+        add_address(iface, IpNetwork::V6(ipv6), handle.clone()).await
+    {
+        warn!("Error adding address to iface. Error: {}", e);
+        return;
+    }
+    if let Err(e) = set_link_up(handle, iface).await {
+        warn!("Error setting link up. Error: {}", e);
+        return;
+    }
+
+    setup_sriov(iface, num_sriov);
+}
+
+pub async fn init_lo_network(handle: rtnetlink::Handle) {
+    let localhost_ipv4: IpNetwork = "127.0.0.1/8".parse().unwrap();
+    let localhost_ipv6: IpNetwork = "::1/128".parse().unwrap();
+
+    if let Err(e) = add_address("lo", localhost_ipv4, handle.clone()).await {
```

Comment:
move "lo" to constant at top of file as we will likely have multiple loopback names in the future